data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeInfo.avsc

```diff
@@ -47,6 +47,11 @@
         "logicalType": "timestamp-millis"
       },
       "doc": "Time ([Instant]) in milliseconds when the record was updated or added."
+    },
+    {
+      "name": "virtualNodeState",
+      "type": ["string"],
```

Comment:
The AVRO schema doesn't define storage but communications (I appreciate, an in a compacted topic, they persist).
This defines the API, so I think it's important to signal what the accepted values are. I think we use enum elsewhere in the AVRO messages in cases like this (e.g. `FlowStatus`).

Reply:
I felt that it would be safer to start with `string`, as at this time we are not even sure which states we are going to have. Also, it is quite likely that we might need to add more states in the near future.
If we go with `enum` right from the start then there is a high chance that we might need to add more values soon, which would mean that we will have to make changes to `corda-api` and release a new version, which is always a hassle.

My advise would be to start with `string` and switch to `enum` when this functionality matures. This is an internal cluster API anyway and end users should not be affected.